### PR TITLE
[SPARK-3873] [build] Enable import ordering error checking.

### PR DESCRIPTION
**Pull request by vanzin**
Turn import ordering violations into build errors, plus a few adjustments
to account for how the checker behaves. I'm a little on the fence about
whether the existing code is right, but it's easier to appease the checker
than to discuss what's the more correct order here.

Plus a few fixes to imports that cropped in since my recent cleanups.

### CODE REVIEW

**Comment by vanzin** [comment]
I'll let this simmer for a few days to wait for new PRs that might have violations to go in, and fix those... but at some point I'll get bored of fixing things and probably push this.

**Comment by vanzin** [comment]
retest this please

File: scalastyle-config.xml [hunk 1]
```diff
@@ -187,6 +187,17 @@ This file is divided into 3 sections:
     scala.collection.JavaConverters._ and use .asScala / .asJava methods</customMessage>
   </check>
 
+  <!-- Import ordering. Currently warning only since there are lots of violations. -->
```
Comment (rxin) [inline scalastyle-config.xml]: you'd need to remove this line

**Comment by rxin** [comment]
LGTM.

**Comment by rxin** [comment]
You sh

**Comment by vanzin** [comment]
Will probably do that Monday morning (or Sunday evening if I'm feeling bored).

**Comment by rxin** [comment]
I've merged this. Thanks.

**Pull request closed without merging**